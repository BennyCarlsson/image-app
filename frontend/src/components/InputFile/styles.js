import styled from 'styled-components'

export const Label = styled.label`
  display: flex;
  flex-direction: column;
  justify-content: flex-end;
  align-items: center;
  width: 100%;
  height: 100%;
  z-index: 1;
`

export const Input = styled.input`
  position: absolute;
  width: 100%;
  height: 100%;
  opacity: 0;
  z-index: 1;
`

export const Text = styled.p`
  color: white;
  font-weight: 700;
  font-size: 2rem;
  text-shadow: 0px 1px 3px #484848;
  z-index: 1;
`
export const Button = styled.span`
  background-color: #50cdb1;
  color: #fff;
  font-family: Lato, sans-serif;
  font-weight: 700;
  font-size: 1rem;
  padding: 0.8rem 3rem;
  -webkit-appearance: none;
  appearance: none;
  border-radius: 25rem;
  border: none;
  margin-top: 2.4rem;
  cursor: pointer;
  transition: background-color 0.2s;
  display: flex;
  justify-content: center;
  align-items: center;
  margin-bottom: 48px;
  z-index: 2;
`
