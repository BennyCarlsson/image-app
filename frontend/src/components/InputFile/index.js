import React from 'react'
import { Button, Input, Label, Text } from './styles'

const InputFile = ({ handleFile, value }) => {
  return (
    <Label>
      <Text>Select a file or drag here</Text>
      <Input
        type='file'
        accept='image/jpeg, image/png'
        value={value}
        onChange={(e) => {
          handleFile(e)
        }}
      />
      <Button>Upload Image</Button>
    </Label>
  )
}

export default InputFile
