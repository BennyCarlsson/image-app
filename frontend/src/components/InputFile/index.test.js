import React from 'react'
import { render, screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import InputFile from '.'

describe('<InputFile />', () => {
  const mockHandleFile = jest.fn()
  const file = new File(['test'], 'test.jpeg', {
    type: 'image/jpeg',
  })

  it('should match snapshot', () => {
    const { asFragment } = render(<InputFile handleFile={mockHandleFile} />)
    expect(asFragment()).toMatchSnapshot()
  })

  it('should render without crashing', () => {
    render(<InputFile handleFile={mockHandleFile} />)
    expect(screen).toBeTruthy()
  })

  it('should upload input file', () => {
    render(<InputFile handleFile={mockHandleFile} />)
    const input = screen.getByLabelText(/Select a file or drag here/i)
    userEvent.upload(input, file)
    expect(input.files[0]).toStrictEqual(file)
    expect(input.files.item(0)).toStrictEqual(file)
    expect(input.files).toHaveLength(1)
  })

  it('should call handleFile', () => {
    render(<InputFile handleFile={mockHandleFile} />)
    const input = screen.getByLabelText(/Select a file or drag here/i)
    expect(mockHandleFile).not.toBeCalled()
    userEvent.upload(input, file)
    expect(mockHandleFile).toBeCalledTimes(1)
  })
})
