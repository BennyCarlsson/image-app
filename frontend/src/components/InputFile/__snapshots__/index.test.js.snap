// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InputFile /> should match snapshot 1`] = `
<DocumentFragment>
  <label
    class="sc-bdvvtL dyeFrr"
  >
    <p
      class="sc-dkPtRN jIjpcz"
    >
      Select a file or drag here
    </p>
    <input
      accept="image/jpeg, image/png"
      class="sc-gsDKAQ eBsTib"
      type="file"
      value=""
    />
    <span
      class="sc-hKwDye dyjigE"
    >
      Upload Image
    </span>
  </label>
</DocumentFragment>
`;
